### PR TITLE
[dv,i2c] Add I2C fcov covergroups

### PR DESCRIPTION
**Pull request by hcallahan-lowrisc**
Some cleanup and adding covergroups.

Closes: https://github.com/lowRISC/opentitan/issues/17447

### CODE REVIEW

File: hw/ip/i2c/dv/env/i2c_env_cov.sv
Comment (crteja) [inline hw/ip/i2c/dv/env/i2c_env_cov.sv:104]: Both DUT and TB can not be in same mode. These bins might not be required

File: hw/ip/i2c/dv/env/i2c_env_cov.sv
Comment (crteja) [inline hw/ip/i2c/dv/env/i2c_env_cov.sv:211]: Shouldn't this also have a `binsof(read) intersect {0}` similar to `read_address_byte` bin

File: hw/ip/i2c/dv/env/i2c_env_cov.sv
Comment (crteja) [inline hw/ip/i2c/dv/env/i2c_env_cov.sv:214]: `cp_rcont` can be used to check for `read_with_ack` and `read_with_nack`.
If in a read packet, `rcont` is set to 1, DUT will issue an ACK, else NACK will be issued at the end of the packet

File: hw/ip/i2c/dv/env/i2c_env_cov.sv
Comment (crteja) [inline hw/ip/i2c/dv/env/i2c_env_cov.sv:260]: In target mode read transactions, ACK/NAK signal is updated in abyte[0]. It would be good to have a cross for ACK/NAK and RSTART/STOP
image

File: hw/ip/i2c/dv/env/i2c_env_cov.sv
Comment (crteja) [inline hw/ip/i2c/dv/env/i2c_env_cov.sv:305]: This covergroup is implemented in https://github.com/lowRISC/opentitan/pull/17849, These comments can be removed.

File: hw/ip/i2c/dv/env/i2c_env_cov.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/i2c_env_cov.sv]: updated in [054ed75](https://github.com/lowRISC/opentitan/pull/18003/commits/054ed75a3dfe38fb9fe96fca752289ad12f53459)

File: hw/ip/i2c/dv/env/i2c_env_cov.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/i2c_env_cov.sv]: updated in [054ed75](https://github.com/lowRISC/opentitan/pull/18003/commits/054ed75a3dfe38fb9fe96fca752289ad12f53459)

File: hw/ip/i2c/dv/env/i2c_env_cov.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/i2c_env_cov.sv]: updated in [054ed75](https://github.com/lowRISC/opentitan/pull/18003/commits/054ed75a3dfe38fb9fe96fca752289ad12f53459)

File: hw/ip/i2c/dv/env/i2c_env_cov.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/i2c_env_cov.sv]: updated in [054ed75](https://github.com/lowRISC/opentitan/pull/18003/commits/054ed75a3dfe38fb9fe96fca752289ad12f53459)

**Review by crteja — COMMENTED** [review]
with changes in [054ed75](https://github.com/lowRISC/opentitan/pull/18003/commits/054ed75a3dfe38fb9fe96fca752289ad12f53459), implementation looks good to me.

**Comment by marnovandermaas** [comment]
@hcallahan-lowrisc do you mind having a look at this? If you give the OK, I'll approve and merge it.

**Comment by hcallahan-lowrisc** [comment]
Removed comments of yet-to-be-implemented cg's, follow up issue to implement these is here : TODO.

**Review by marnovandermaas — APPROVED** [review]
Ok, seems like we're happy with this. Let's wait for CI to finish before merging.

**Pull request merged**
